Week-7/Exercise-7.3/index.js

```diff
@@ -0,0 +1,65 @@
+// DETECT A LOOP IN SLL
+class SLLNode {
+  constructor(value) {
+    this.value = value;
+    this.next = null;
+  }
+}
+
+let head;
+
+function createSLL(arr) {
+  if (arr.length === 0) return;
+  head = new SLLNode(arr[0]);
+
+  let current = head;
+  for (let i = 1; i < arr.length; i++) {
+    current.next = new SLLNode(arr[i]);
+    current = current.next;
+  }
+}
+
+function traverseSLL() {
+  let current = head;
+  while (current) {
+    console.log(current.value);
+    current = current.next;
+  }
+}
+
+function createALoopInSLL(k) {
+  let current = head;
+  let pivot;
+
+  while (current.next) {
+    current = current.next;
+    k--;
+    if (k == 0) {
+      pivot = current;
+    }
+  }
+
+  current.next = pivot;
+}
+
+function isLoopExistInSLL() {
+  let sp = (fp = head);
+
+  do {
+    fp = fp.next;
+    if (fp) fp = fp.next;
+
+    sp = sp.next;
+  } while (fp && sp != fp);
+
+  if (fp == null) return console.log(false);
+
+  return console.log(true);
```

Comment:
@apandyamob why return console.log(true). Is it just return true ?

Reply:
yes @RajeshSivanesan  just for the sake of printing I have taken the console log within the function. 

And intentionally it is true if there is no loop detected before executing this line function would be returned with false